src/merge.py

```
@@ -807,277 +401,196 @@ def main():
+        if not validate_2fa(icloud_service):
             term.print_failed()
-            print_step(TAG_CANCEL, f"failed to remove events for {cancel_cleanup_date}: {err}", True)
-            if not dry_run:
-                send_telegram_message(f"⚠️ Cancel cleanup failed for {cancel_cleanup_date}: {err}")
-
-    lifecycle_changed = _handle_override_date_lifecycle(state, today)
-    ingest_changed = _ingest_override_flag(args, telegram_commands, state)
-    compute_changed = _compute_and_persist_override_date(args, state, skip_days, today)
-    if not dry_run and (offset_changed or lifecycle_changed or ingest_changed or compute_changed):
-        _save_state(state_path, state)
+            raise RuntimeError("2FA validation failed")
```

Ruff (0.15.6)

[warning] Abstract `raise` to an inner function

(TRY301)